packages/core/src/logger.ts

```diff
@@ -107,7 +116,7 @@ export class Logger {
         level: LogLevelNames[level],
         timestamp: new Date().toISOString(),
         msg: this.prefix ? `${this.prefix}${msg}` : msg,
-        ...data,
+        ...(data && typeof data === 'object' ? data : { data }),
```

Comment:
If data is an array this will still merge it into the main object:
<img width="600" alt="Screenshot 2025-04-02 at 9 11 06 AM" src="https://github.com/user-attachments/assets/f3fa4aef-63bd-4656-9a39-f42cc33306dc" />